### ISSUE TITLE
heap-use-after-free at bvconst_tst_bit terms/bv_constants.c:354 (mcsat-assumptions) 

### ISSUE THREAD

**Issue by rainoftime** [SRI-CSL/yices2#270]
Hi, for the following formula
~~~~
(set-option :produce-unsat-model-interpolants true)
(set-logic QF_UFBV)
(declare-const bv_12-0 (_ BitVec 12))
(declare-const bv_14-0 (_ BitVec 14))
(declare-const bv_25-0 (_ BitVec 25))
(declare-const bv_16-0 (_ BitVec 16))
(declare-const bv_24-0 (_ BitVec 24))
(declare-const bv_21-0 (_ BitVec 21))
(declare-const bv_11-0 (_ BitVec 11))
(declare-const bv_18-0 (_ BitVec 18))
(declare-const bv_25-1 (_ BitVec 25))
(declare-const bv_31-0 (_ BitVec 31))
(declare-const bv_9-4 (_ BitVec 9))
(declare-const bv_20-0 (_ BitVec 20))
(declare-const bv_15-0 (_ BitVec 15))
(declare-const bv_7-9 (_ BitVec 7))
(assert (or (bvugt ((_ rotate_left 1) (concat ((_ extract 6 3) bv_16-0) ((_ repeat 8) #b01010010101))) (concat ((_ extract 6 3) bv_16-0) ((_ repeat 8) #b01010010101))) (bvult #x57 #x57) (distinct bv_14-0 bv_14-0 bv_14-0 bv_14-0)))
(check-sat-assuming-model (bv_7-9 bv_9-4 bv_11-0 bv_12-0 bv_14-0 bv_15-0 bv_16-0 bv_18-0 bv_20-0 bv_21-0 bv_24-0 bv_25-0 bv_25-1 bv_31-0 ) ((_ bv80 7) (_ bv7 9) (_ bv379 11) (_ bv1309 12) (_ bv3654 14) (_ bv21115 15) (_ bv46314 16) (_ bv204014 18) (_ bv630646 20) (_ bv91859 21) (_ bv9869541 24) (_ bv16221776 25) (_ bv8021998 25) (_ bv427384175 31) ))
(get-unsat-model-interpolant)
~~~~

yices2 b6f1b5b
~~~~
unsat
(bv-lt
 (bv-concat (bv-extract 6 3 bv_16-0)=================================================================
==2959==ERROR: AddressSanitizer: heap-use-after-free on address 0x60200000cd98 at pc 0x000000646f0b bp 0x7ffe35af7790 sp 0x7ffe35af7780
READ of size 4 at 0x60200000cd98 thread T0
    #0 0x646f0a in bvconst_tst_bit terms/bv_constants.c:354
    #1 0x713ea2 in string_buffer_append_bvconst utils/string_buffers.c:229
    #2 0x499a02 in build_bv io/yices_pp.c:296
    #3 0x49a867 in get_string io/yices_pp.c:470
    #4 0x841f13 in get_string io/pretty_printer.c:217
    #5 0x8427f6 in print_atomic io/pretty_printer.c:383
    #6 0x84398f in print_atomic_token io/pretty_printer.c:618
    #7 0x8460c2 in print_token io/pretty_printer.c:1020
    #8 0x847ee9 in flush_tokens io/pretty_printer.c:1388
    #9 0x8483bf in flush_wide_blocks io/pretty_printer.c:1458
    #10 0x848f89 in process_token io/pretty_printer.c:1582
    #11 0x849243 in pp_push_token io/pretty_printer.c:1642
    #12 0x49b2c4 in pp_string io/yices_pp.c:684
    #13 0x49410d in pp_term_recur io/term_printer.c:2676
    #14 0x492cef in pp_bit_array io/term_printer.c:2445
    #15 0x492e0e in pp_bvarray_term io/term_printer.c:2453
    #16 0x493c0b in pp_term_idx io/term_printer.c:2611
    #17 0x4941a9 in pp_term_recur io/term_printer.c:2682
    #18 0x490aad in pp_binary_atom io/term_printer.c:2009
    #19 0x493908 in pp_term_idx io/term_printer.c:2572
    #20 0x4941a9 in pp_term_recur io/term_printer.c:2682
    #21 0x494247 in pp_term_full io/term_printer.c:2707
    #22 0x7ec725 in show_unsat_model_interpolant frontend/smt2/smt2_commands.c:4174
    #23 0x7ee94e in smt2_get_unsat_model_interpolant frontend/smt2/smt2_commands.c:4808
    #24 0x803786 in eval_smt2_get_unsat_model_interpolant frontend/smt2/smt2_term_stack.c:852
    #25 0x4e7e93 in tstack_eval parser_utils/term_stack2.c:5434
    #26 0x7fb459 in smt2_parse frontend/smt2/smt2_parser.c:316
    #27 0x7fcb94 in parse_smt2_command frontend/smt2/smt2_parser.c:960
    #28 0x408c69 in main frontend/yices_smt2.c:739
    #29 0x7fa9f668a82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)

~~~~

**Comment by BrunoDutertre** [comment]
That's unrelated to solving with assumptions. I'll fix it.

**Comment by BrunoDutertre** [comment]
@dddejan: this is fixed in master. Is it OK to merge master in this branch?

**Comment by dddejan** [comment]
> @dddejan: this is fixed in master. Is it OK to merge master in this branch?

Yes, master is always welcome

**Comment by BrunoDutertre** [comment]
OK. Fixed in cd907fd1f15950b9705acfca392a9258b169d529.
I'm closing this.
